src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,270 @@
+    AllocatedArray<std::byte> block;
+
+    // Try up to 5 times before giving up
+    for (uint8_t attempt = 0; attempt < 5; attempt++) {
+      success = GetFlightLogBlock(block, flight_id, block_id, env);
+      if (success)
+        break;
+    }
+
+    if (!success)
+      break;
+
+    bos.Write(block);
+
+    // Make sure that bytes remaining cannot underflow
+    if (block.size() > bytes_remaining)
+      return false;
+
+    bytes_remaining -= block.size();
+
+    // Progress in percents downloaded
+    float progress =
+      100.0f * (1.0f - static_cast<float>(bytes_remaining) /
+                         static_cast<float>(flight.internal.lx_eos.file_size));
+    env.SetProgressPosition(static_cast<unsigned int>(floor(progress)));
+  }
+
+  port.Flush();
+
+  if (success) {
+    bos.Flush();
+    fos.Commit();
+  }
+
+  env.SetProgressPosition(100);
+
+  port.StartRxThread();
+  return success;
+}
+
+bool
+LXEosDevice::GetNumberOfFlights(uint8_t& flight_count,
+                                OperationEnvironment& env)
+{
+  EosGetNumOfFlights data;
+  std::byte response[3];
+  response[0] = std::byte{ 0x06 };
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  try {
+    port.FullRead({ response + 1, sizeof(response) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Check CRC
+  if (CalculateCRC(response, sizeof(response), 0xFF) != 0x00)
+    return false;
+
+  flight_count = std::to_integer<uint8_t>(response[1]);
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightInfo(const uint8_t index,
+                           RecordedFlightInfo& flight,
+                           OperationEnvironment& env)
+{
+  if (index == 0) // Invalid
+    return false;
+
+  EosRequestFlightInfo data;
+  data.flight_id = index;
+  std::byte response[94];
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  data.crc = CalculateCRC(data_array, sizeof(data) - 1, 0xFF);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+  response[0] = std::byte{ 0x06 };
+
+  try {
+    port.FullRead({ response + 1, sizeof(response) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  if (CalculateCRC(response, sizeof(response), 0xFF) != 0x00) // Check CRC
+    return false;
+
+  /* Flight ID for downloading is the index (1 = newest),
+  not the ID from this message */
+  flight.internal.lx_eos.flight_id = index;
+
+  uint32_t julianDate, time_takeoff, time_landing, file_size;
+  std::memcpy(&julianDate, &response[13], 4);
+  std::memcpy(&time_takeoff, &response[17], 4);
+  std::memcpy(&time_landing, &response[21], 4);
+  std::memcpy(&file_size, &response[89], 4);
```

Comment:
Byte order? Why are you using a byte array with hard-coded offsets here, instead of declaring a (documented) struct, as you did in other parts of this PR?